### ISSUE TITLE
Scope of receiving coins tab might not be obvious

### ISSUE THREAD

**Issue by dexX7** [OmniLayer/omnicore#139]
[On Reddit](https://www.reddit.com/r/omni/comments/3dioag/the_omnicore_windows_offline_client/) /u/JuicyGrabs [wrote](https://www.reddit.com/r/omni/comments/3db6bn/state_of_the_layer_all_hands_july_14_2015/ct4rj5w):

> The Receive section also needs a remake. You can't have two Send Omni / Send Bitcoin tabs in Send area and just one Receive Bitcoin tab in the Receive area. It confuses the user.

> You might want to consider new/different input fields in Receive menu. It's confusing to an average user to be able to send Omni coins or other coins and only be able to receive Bitcoin according to the interface.

Right now I'm not really sure how to tackle this, given that I personally never use the receive section, except for generating new addresses, but let's try to brainstorm a bit.

**Comment by zathras-crypto** [comment]
Interesting feedback, I'm the same I don't use the receive dialog, that's more for generating "requests for payment" I thought, but if it has the potential to cause confusion perhaps we should look at a way to do a request for payment in an Omni token - I'll take a look :)

**Comment by JuicyGrabs** [comment]
I guess it's best to see things from the perspective of someone who's not in the know so to speak. 

Imagine a user who just wants to use this wallet to store, send/receive and trade Maidsafecoin for example. Said user might not even own any Bitcoin and he's only interested in playing around with Maidsafecoin. When he opens up the wallet he will see the send tab which would probably include a list of all the coins available on Omni platform. Imagine the confusion if the receive address displays a receive Bitcoin address instead of a Maidsafe address. The user shouldn't know what's behind the hood so to speak and how the send/receive mechanics work. He would probably just think something along these lines : "WTF, I don't want to receive BTC. What is this shit?"

**Comment by dexX7** [comment]
Very good points.

> The send option should only display these two coins and perhaps also feature an expand button to reveal empty balances.

Agreed, this would be great.

> Receive tab should only include coins currently held in the wallet.

> receive address separately based on what coins the user holds

So basically you suggest that there should be some kind of categorization of the addresses generated via the receiving tab, like "Bitcoin", "MaidSafeCoin", ..., and when an user creates a new address, then he or she should be able to select a category (= the token to receive)?

**Comment by JuicyGrabs** [comment]
Yes, something like that.

**Comment by zathras-crypto** [comment]
>>> Say a user has 100 Maidsafecoins and 50 Omni/Mastercoins. The send option should only display these two coins and perhaps also feature an expand button to reveal empty balances. 

>> The send option should only display these two coins and perhaps also feature an expand button to reveal empty balances.

> Agreed, this would be great.

Can I just clarify something here guys, are you saying that you get a list of all tokens in the send dialog, even those where you don't have any balances?  If so that's a bug & shouldn't occur.

There is code within ```sendmpdialog.cpp``` that should ensure that tokens are added conditionally to the property selector, only if >0 balance for that property id...

```
if ((global_balance_money[propertyId] > 0) || (global_balance_reserved[propertyId] > 0)) {
```

Apologies if I've missed the point here, and the feedback regarding the receive tab is appreciated & will be addressed :)

**Comment by dexX7** [comment]
> Can I just clarify something here guys, are you saying that you get a list of all tokens in the send dialog, even those where you don't have any balances? If so that's a bug & shouldn't occur.

Correct, addresses with zero tokens are hidden. I could see some value to have a checkbox to hide addresses with zero BTC though, if this doesn't require endless wallet queries. I could be biased by the regtests though, given that a long list of addresses is usually the last state. On the other hand, if there were such a feature, addresses with tokens may be overlooked, only because the filter is active.

**Comment by zathras-crypto** [comment]
> I could see some value to have a checkbox to hide addresses with zero BTC though, if this doesn't require endless wallet queries.

The difficulty here is that we don't maintain an internal tally of BTC balances per address, and the wallet maintains BTC balances by way of outputs, rather than by address.  What's the value of hiding addresses with zero BTC in the context of sending Omni tokens?  Just that an address without BTC doesn't have sufficient fees for a send?  

I'm sure we could do it, especially since I want to take us in the general direction of BTC balances per address to enable the display of BTC as an entry in the "Balances" page (Bitcoin Core as of 0.11 still doesn't provide an easy to use method to get an address balance, so we could contribute something upstream at least on the RPC side) and simplify code around fee warnings and such, but I'm not sure I agree we *should* do offer filtering of empty BTC balances when looking at the context of Omni balances, mainly because of the potential for user confusion.
